src/rcc/reset_reason.rs

```diff
@@ -0,0 +1,139 @@
+//! A module that can capture the RCC registers and read the reason why the mcu
+//! has reset
+
+use core::fmt::Display;
+
+/// Gets and clears the reason of why the mcu was reset
+#[rustfmt::skip]
+#[allow(dead_code)]
+pub fn get_reset_reason(rcc: &mut crate::stm32::RCC) -> ResetReason {
+    let reset_reason = rcc.rsr().read();
+
+    // Clear the register
+    rcc.rsr().modify(|_, w| w.rmvf().reset());
+
+    #[cfg(feature = "rm0492")]
+    // See R0492 Section 10.3.4 Reset source identification
+    return match (
+        reset_reason.lpwrrstf().is_reset_occurred(),
+        reset_reason.wwdgrstf().is_reset_occurred(),
+        reset_reason.iwdgrstf().is_reset_occurred(),
+        reset_reason.sftrstf().is_reset_occurred(),
+        reset_reason.pinrstf().is_reset_occurred(),
+        reset_reason.borrstf().is_reset_occurred(),
+    ) {
+        (false, false, false, false, true, true) => {
+            ResetReason::PowerOnReset
+        }
+        (false, false, false, false, false, true) => {
+            ResetReason::PinReset
+        }
+        (false, false, false, true, false, true) => {
+            ResetReason::SystemReset
+        }
+        (false, true, false, false, false, false) | (false, true, false, false, false, true) => {
+            ResetReason::WindowWatchdogReset
+        }
+        (false, false, true, false, false, false) => {
+            ResetReason::IndependentWatchdogReset
+        }
+        (true, false, false, false, false, false) => {
```

Comment:
These don't quite agree with the table in section 10.3.4. They won't match if PINRSTF is set. Is this deliberate?

Reply:
Not deliberate. I think there are a few copy-pasta errors from the H7 codebase in this table.